native/cocos/platform/android/AndroidPlatform.cpp

```diff
@@ -430,10 +454,19 @@ int AndroidPlatform::getSdkVersion() const {
     return AConfiguration_getSdkVersion(_app->config);
 }
 
-int32_t AndroidPlatform::run(int argc, const char **argv) {
-    if (cocos_main(argc, argv) != 0) {
-        return -1;
-    }
+int32_t AndroidPlatform::run(int  /*argc*/, const char **argv) {
+    auto *app = reinterpret_cast<struct android_app *>(argv);
+    cc::FileUtilsAndroid::setassetmanager(app->activity->assetManager);
+    _app = app;
+    CC_SAFE_DELETE(_inputProxy);
+    _inputProxy = ccnew GameInputProxy(this);
+    _inputProxy->setLaunched(_isInited);
+    _app->userData = _inputProxy;
+    _app->onAppCmd = handleCmdProxy;
```

Comment:
It is better to put these logic into `init()`
```c++
auto *app = reinterpret_cast<struct android_app *>(argv);
    cc::FileUtilsAndroid::setassetmanager(app->activity->assetManager);
    _app = app;
    _inputProxy = ccnew GameInputProxy(this);
    _inputProxy->setLaunched(_isInited);
    _app->userData = _inputProxy;
    _app->onAppCmd = handleCmdProxy;
```

Reply:
Then add a `destroy()`, and delete `_inputProxy` and unregister interfaces in it.